models/spend_daily.sql

```diff
@@ -0,0 +1,193 @@
+    select
+        metering_history.start_time::date as usage_date,
+        'serverless' as usage_type,
+        'snowpipe' as usage_type_subcategory_1,
+        sum(metering_history.credits_used * daily_rates.effective_rate) as spend_usd
+    from snowflake.account_usage.metering_history
+    left join {{ ref('daily_rates') }} on
+        metering_history.start_time::date = daily_rates.date
+        and daily_rates.usage_type = 'cloud services'
+        and daily_rates.service_type = 'COMPUTE'
+    where metering_history.service_type = 'PIPE'
+    group by 1, 2, 3
+),
+
+query_acceleration_spend_daily as (
+    select
+        metering_history.start_time::date as usage_date,
+        'serverless' as usage_type,
+        'query acceleration' as usage_type_subcategory_1,
+        sum(metering_history.credits_used * daily_rates.effective_rate) as spend_usd
+    from snowflake.account_usage.metering_history
+    left join {{ ref('daily_rates') }} on
+        metering_history.start_time::date = daily_rates.date
+        and daily_rates.usage_type = 'cloud services'
+        and daily_rates.service_type = 'COMPUTE'
+    where metering_history.service_type = 'QUERY_ACCELERATION'
+    group by 1, 2, 3
+),
+
+replication_spend_daily as (
+    select
+        metering_history.start_time::date as usage_date,
+        'serverless' as usage_type,
+        'replication' as usage_type_subcategory_1,
+        sum(metering_history.credits_used * daily_rates.effective_rate) as spend_usd
+    from snowflake.account_usage.metering_history
+    left join {{ ref('daily_rates') }} on
+        metering_history.start_time::date = daily_rates.date
+        and daily_rates.usage_type = 'cloud services'
+        and daily_rates.service_type = 'COMPUTE'
+    where metering_history.service_type = 'REPLICATION'
+    group by 1, 2, 3
+),
+
+search_optimization_spend_daily as (
+    select
+        metering_history.start_time::date as usage_date,
+        'serverless' as usage_type,
+        'search optimization' as usage_type_subcategory_1,
+        sum(metering_history.credits_used * daily_rates.effective_rate) as spend_usd
+    from snowflake.account_usage.metering_history
+    left join {{ ref('daily_rates') }} on
+        metering_history.start_time::date = daily_rates.date
+        and daily_rates.usage_type = 'cloud services'
+        and daily_rates.service_type = 'COMPUTE'
+    where metering_history.service_type = 'SEARCH_OPTIMIZATION'
+    group by 1, 2, 3
+)
+
+select * from storage_spend_daily
+union all
+select * from compute_spend_daily
+union all
+select * from cloud_services_spend_daily
+union all
+select * from automatic_clustering_spend_daily
+union all
+select * from materialized_view_spend_daily
+union all
+select * from snowpipe_spend_daily
+union all
+select * from query_acceleration_spend_daily
+union all
+select * from replication_spend_daily
+union all
+select * from search_optimization_spend_daily
+union all
+select * from storage_spend_daily
```

Comment:
thinking out loud: i wonder if it would be useful to guarantee 1 row per day per usage type? 

if you plug this model straight into a BI tool, I think it would be useful to have 0 on certain days for a certain usage type, rather than a missing value.  thoughts?